rust/headless-client/src/lib.rs

```diff
@@ -451,10 +451,13 @@ async fn ipc_listen() -> Result<std::convert::Infallible> {
             .next_client()
             .await
             .context("Failed to wait for incoming IPC connection from a GUI")?;
-        Handler::new(stream)?
+        if let Err(error) = Handler::new(stream)?
```

Comment:
can we make `run` infallible and log within it? That way we also only loop here when we want the handler to explicitly terminate